texastribune/tt-docker-base
# Add version bumping script

#### What's this PR do?

Adds a python CLI tool to make it easier to bump the version by executing the command: `make bump`.

Answering the prompts will allow a user to:
- Change to master branch and pull latest (as recommended by our README)
- Change major/minor/patch version number in ~/VERSION file in the active branch
- Save and commit the new ~/VERSION file.
- Add a tag to the commit and push the tag and active branch to remote.

#### Why are we doing this? How does it help us?

Updating the version manually is relatively quick task but it's also a chore and we should automate chores when we can. 
 Also, if you're new to the team or you haven't dealt with tt-docker-base in a while, it's easy to forget the steps involved in this process.

#### How should this be manually tested?

1) Because this script manipulates git history, it's important you create a new branch for testing purposes. Fetch this branch, check it out, and then create a new branch from it `git checkout -b bump-test'
2) Run `make bump` 
3) The first prompt will show a warning and ask whether you wish to switch to the master branch. For testing purposes, enter 'no' or 'n' to stay on your testing branch.
4) Follow the prompts to completion and double check the following:
- ~/VERSION file has changed as desired in local and your remote branch.
- Change to ~/VERSION file is recorded in git history in local and your remote branch.
- Tag has been pushed to remote

Clean up:
- If you pushed your testing branch to remote, you'll want to delete it to keep our remote repo clean
- Since git tags are not branch specific, you'll also need to clean up any tags you pushed to remote. You can delete a tag locally like so `git tag -d <tagname>` eg. `git tag -d 2.1.0`. You can then delete it from the remote with the following: `git push --delete origin 2.1.0`

#### How should this change be communicated to end users?

#### Are there any smells or added technical debt to note?

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwKqLrNFM17Lq15R/recfh5zHIfb6s19D9?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### TODOs / next steps:
